src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -158,7 +161,16 @@ private function collectPaths(ApiResource $resource, ResourceMetadataCollection
 
             [$requestMimeTypes, $responseMimeTypes] = $this->getMimeTypes($operation);
 
-            $operationId = $operation->getOpenapiContext()['operationId'] ?? $this->normalizeOperationName($operationName);
+            $operationId = $hasOpenapi && $openapiOperation->getOperationId() ? $openapiOperation->getOperationId() : $this->normalizeOperationName($operationName);
```

Comment:
I'd like to find something better to avoid these calls, what about declaring the default operation line 140:

```
if (!$openapiOperation) {
    $openapiOperation = new Model\Operation(operationId: $this->normalizeOperationName($operationName));
}
```

Then you can just use `$openapiOperation->getOperationId();`